readthedocs/gold/views.py

```diff
@@ -152,3 +139,149 @@ def post(self, request, *args, **kwargs):
         gold_user.projects.remove(project)
 
         return HttpResponseRedirect(self.get_success_url())
+
+
+class GoldCreateCheckoutSession(GenericView):
+
+    http_method_names = ['post']
+
+    def post(self, request, *args, **kwargs):
+        try:
+            user = request.user
+            schema = 'https' if settings.PUBLIC_DOMAIN_USES_HTTPS else 'http'
+            url = reverse('gold_subscription')
+            url = f'{schema}://{settings.PRODUCTION_DOMAIN}{url}'
+            price = json.loads(request.body).get('priceId')
+            log.info('Creating Stripe Checkout Session. user=%s price=%s', user, price)
+            checkout_session = stripe.checkout.Session.create(
+                client_reference_id=user.username,
+                customer_email=user.emailaddress_set.filter(verified=True).first() or user.email,
+                payment_method_types=['card'],
+                line_items=[
+                    {
+                        'price': price,
+                        'quantity': 1,
+                    }
+                ],
+                mode='subscription',
+                # We use the same URL to redirect the user. We only show a different notification.
+                success_url=url,
+                cancel_url=url,
+            )
+            return JsonResponse({'session_id': checkout_session.id})
+        except:  # noqa
+            log.exception('There was an error connecting to Stripe.')
+            return JsonResponse(
+                {
+                    'error': 'There was an error connecting to Stripe.'
+                },
+                status=500,
+            )
+
+
+class GoldSubscriptionPortal(GenericView):
+    http_method_names = ['post']
+
+    def post(self, request, *args, **kwargs):
+        user = request.user
+
+        # TODO: review if User <-> GoldUser being ``ForeignKey`` makes
+        # sense no that we are removing the one time donation
```

Comment:
```suggestion
        # sense now that we are removing the one time donation
```

Do we need to remove one-time donations? I feel like they are useful, but it might also just make more sense to direct those people to GH sponsors or OpenCollective or something? If possible, we should probably keep it tho, because some users want a one-time way of paying to remove ads.

Reply:
We _don't have to_. I know that we decided to remove one-time payments, but I'm not sure if it was because it was too complex to maintain or we wanted to removed completely for other reasons. If we want to keep it, we may not be able re-use the code from `Subscription` (in the medium term) or we may need to do something extra to mark them as `subscribed=False` (as the current `GoldUser` is doing). However, with the latest changes to our Gold Membership, I'm not sure if doing a one-time donation has extra benefits for the user, so we may not need to track this in the database. IIRC to remove ads you need to be subscribed and pay USD 5 (user) or 50 (company) per project.

Reply:
Hrm yea -- I think in the past we've done one-time donations them manually marked projects as Ad-free. That should still work fine. 

It would be useful to continue supporting one-time donations. We support it here: https://readthedocs.org/sustainability/ -- maybe that code will still "just work", because it isn't tied to the Gold stuff?

Reply:
> I'd prefer if the form submitted to Stripe and you put your CC info directly there, so we don't ever have a chance to see it, but probably fine to keep the existing workflow if it still works.

The idea is to remove (or replace with Stripe Checkout) this workflow completely. They are not going to be included in the new theme.

Reply:
Yea, I think we need a way to do one-time payments then. It doesn't need to be hooked up to the DB at all I don't think. I'm 👍  on using another platform for this, so we don't need to maintain code. If we can easily do it with Stripe, that would be great, but I don't care too much.